From 5c179985028d68323cb0bf349e4268a3aa621035 Mon Sep 17 00:00:00 2001
From: Stanislav Vlasic <svlasic@gmail.com>
Date: Thu, 20 Feb 2014 06:45:42 +0100
Subject: [PATCH] Add --enable-mali20 as configure option

This option is used for Mali 400 API 20 (r3p2-01rel1) libs.
If not enabled, XBMC configure will report that GLES libs
do not exist.
---
 configure.in | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 815e2db..bbb6769 100644
--- a/configure.in
+++ b/configure.in
@@ -578,6 +578,15 @@ AC_ARG_ENABLE([libav-compat],
 
 ### End of external library options
 
+### Mali 400 API 20 definition (Amlogic)
+AC_ARG_ENABLE([mali20],
+  [AS_HELP_STRING([--enable-mali20],
+  [enable Mali API 20 support (default is no)])],
+  [has_mali20=$enableval],
+  [has_mali20=no])
+
+### End of Mali 400 API 20 definition (Amlogic)
+
 if test "x$host_vendor" != "xapple"; then
   DEFAULT_COMPILE_FLAGS="-fPIC -DPIC -D_REENTRANT"
   DEFAULT_COMPILE_FLAGS="$DEFAULT_COMPILE_FLAGS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64"
@@ -1049,8 +1058,15 @@ if test "$use_gles" = "yes"; then
       AC_MSG_RESULT(== WARNING: OpenGLES support is assumed.)
       LIBS="$LIBS -lEGL -lGLESv2 -lbcm_host -lvcos -lvchiq_arm"
     else
-      AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library))
-      AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library))
+      if test "$has_mali20" = "yes"; then
+	  AC_DEFINE([HAVE_LIBEGL],[1],["Define to 1 if you have the `EGL' library (-lEGL)."])
+	  AC_DEFINE([HAVE_LIBGLESV2],[1],["Define to 1 if you have the `GLESv2' library (-lGLESv2)."])
+	  AC_MSG_RESULT(== WARNING: OpenGLES support is assumed.)
+	  LIBS="$LIBS -lUMP -lEGL -lGLESv2"
+      else
+	  AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library))
+	  AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library))
+      fi
     fi
   fi
 else
-- 
1.9.1

